app/js/arethusa.core/services/api.js

```diff
@@ -86,11 +92,47 @@ angular.module('arethusa.core').service('api', [
 
     /**  
      * navigates application state to supplied sentenceId
+     * making the optional word id(s) selected
      * @param {String} sentenceId 
+     * @param {String[]} wordIds (optional)
      */
-    this.gotoSentence = function(sentenceId) {
+    this.gotoSentence = function(sentenceId, wordIds) {
```

Comment:
You're probably right about this. I struggled with the different behavior of findWord and gotoSentence, since both actually navigate to the sentence.  This is more driven by the limitations of the Arethusa application, which I'm trying to avoid changing as much as possible, then by my use cases.   In Arethusa, the state tokens object only contains the tokens for the current sentence. So you need navigate to the sentence before doing the search for the words.

So I thought about naming the api method gotoSentenceAndFindWord but that seemed ridiculously wordy.

Reply:
additionally, while I could expose a separate API call to select a word, you still have to always navigate to the sentence first, so I'm not sure if it extra message is really worth it.

Reply:
> additionally, while I could expose a separate API call to select a word, you still have to always navigate to the sentence first, so I'm not sure if it extra message is really worth it.

That's interesting, I was imagining that someone would want to navigate to a sentence, highlight a word, then highlight a different word. If `navigator.goTo(sentenceId);` is always called before highlighting, then it makes sense to combine them.

Reply:
As the Arethusa UI already does highlight when you click on a word, a separate api call isn't really necessary for that.